src/main/java/gov/nasa/pds/api/engineering/controllers/MyCollectionsApiController.java

```diff
@@ -125,68 +114,64 @@ public MyCollectionsApiController(ObjectMapper objectMapper, HttpServletRequest
     		
 	
     
-    @SuppressWarnings("unchecked")
 	private Products getProductChildren(String lidvid, int start, int limit, List<String> fields, List<String> sort, boolean onlySummary) throws IOException, LidVidNotFoundException {
-  	
     	if (!lidvid.contains("::")) lidvid = this.productBO.getLatestLidVidFromLid(lidvid);
     	MyCollectionsApiController.log.info("request collection lidvid, collections children: " + lidvid);
-         
-    	try {
-	    	Products products = new Products();
-	    	
-	    	HashSet<String> uniqueProperties = new HashSet<String>();
-	    	
-	      	Summary summary = new Summary();
-	    	
-	    	summary.setStart(start);
-	    	summary.setLimit(limit);
-	    	
-	    	if (sort == null) {
-	    		sort = Arrays.asList();
-	    	}	
-	    	summary.setSort(sort);
-	    	
-	    	products.setSummary(summary);
-	    	
-	    	CollectionProductRefBusinessObject  collectionProductRefBO = new CollectionProductRefBusinessObject(this.esRegistryConnection);
-	    	CollectionProductRelationships collectionProductRelationships = collectionProductRefBO.getCollectionProductsIterable(lidvid, start, limit);
-	    	
-	    	for (EntityProduct eProd : collectionProductRelationships) {
-				  if (eProd != null) {
-		        	MyCollectionsApiController.log.info("request lidvdid: " + eProd.getLidVid() );
-		        	
-		    		Product product = ElasticSearchUtil.ESentityProductToAPIProduct(eProd, this.getBaseURL());
-				    		
-		    		Map<String, XMLMashallableProperyValue> filteredMapJsonProperties = ProductBusinessObject.getFilteredProperties(
-		    				eProd.getProperties(), 
-		    				fields,
-		    				null);
-		
-		    		uniqueProperties.addAll(filteredMapJsonProperties.keySet());
-		
-		    		if (!onlySummary) {
-		        		product.setProperties((Map<String, PropertyArrayValues>)(Map<String, ?>)filteredMapJsonProperties);
-		        		
-		        		products.addDataItem(product);
-		    		}
+
+    	int iteration=0,wsize=0;
+    	HashSet<String> uniqueProperties = new HashSet<String>();
+    	List<String> plidvids = new ArrayList<String>();
```

Comment:
I think an understand of what plidvids and wlidvids are for would also help me to understand the code (longer varaible name ?). I am guessing p is for product but I am missing what w stands for.

Reply:
w - working